cfme/scripting/bz.py

```diff
@@ -0,0 +1,221 @@
+        bz_list.append(BZTestCoverage(id=bug_id, qe_test_coverage=qe_test_coverage))
+
+    return bz_list
+
+
+def cleanup():
+    click.echo("Removing the BZ report file, bz-report.yaml")
+    try:
+        os.remove("bz-report.yaml")
+    except OSError:
+        logger.exception("bz-report.yaml not found")
+
+
+@click.group(help="Functions for generating reports on BZs included in test suite metadata")
+def main():
+    pass
+
+
+@main.command(help="Generate BZ report on BZs that have coverage given a directory")
+@click.argument("directory", default="cfme/tests/")
+def report(directory):
+    get_report(directory)
+
+
+@main.command(help="List open/closed BZs that have test coverage")
+@click.argument("directory", default="cfme/tests/")
+@click.option(
+    "--all",
+    "-a",
+    "bz_status",
+    is_flag=True,
+    help="list all BZs with coverage",
+    default=True,
+    show_default=True,
+    flag_value="all_bzs",
+)
+@click.option(
+    "--open",
+    "-o",
+    "bz_status",
+    is_flag=True,
+    help="list open BZs with coverage",
+    default=False,
+    show_default=True,
+    flag_value="open_bzs"
+)
+@click.option(
+    "--closed",
+    "-c",
+    "bz_status",
+    is_flag=True,
+    help="list closed BZs with coverage",
+    default=False,
+    show_default=True,
+    flag_value="closed_bzs"
+)
+def list(directory, bz_status):
+    info = get_report(directory)
+
+    # get dict of information
+    status = STATUS[bz_status]
+
+    if status["val"]:
+        bz_list = [bug_id for bug_id in info.keys() if info[bug_id]["is_open"] == status["val"]]
+    else:
+        bz_list = [bug_id for bug_id in info.keys()]
+
+    if bz_list:
+        click.echo("The following BZ's {}: \n{}".format(status["text"], "\n".join(bz_list)))
+    else:
+        click.echo("I found no BZ's that {} BZ's".format(status["text"]))
+    cleanup()
+
+
+@main.command(help="Set QE test coverage flag based on automates/coverage metadata")
+@click.argument("directory", default="cfme/tests")
+@click.option(
+    "-s",
+    "--set",
+    "set_bzs",
+    is_flag=True,
+    help="Set QE test coverage on BZs that are marked in coverage/auotmates test metadata",
+    default=False,
+    show_default=True
+)
+@click.option(
+    "--all",
+    "-a",
+    "bz_status",
+    is_flag=True,
+    help="Audit test coverage on all BZs",
+    default=True,
+    show_default=True,
+    flag_value="all_bzs",
+)
+@click.option(
+    "--open",
+    "-o",
+    "bz_status",
+    is_flag=True,
+    help="Audit test coverage on only open BZs",
+    default=False,
+    show_default=True,
+    flag_value="open_bzs"
+)
+@click.option(
+    "--closed",
+    "-c",
+    "bz_status",
+    is_flag=True,
+    help="Audit test coverage on only closed BZs",
+    default=False,
+    show_default=True,
+    flag_value="closed_bzs"
+)
+def coverage(directory, set_bzs, bz_status):
+    info = get_report(directory)
+
+    # get list of open BZs that are open and should have test coverage set
```

Comment:
@john-dupuy I think the comment needs updating since a change has been made to show whatever the user wants

Reply:
updated